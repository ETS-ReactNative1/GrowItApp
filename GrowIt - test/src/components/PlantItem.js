import * as React from 'react';
import { object } from 'prop-types';
import { TouchableOpacity, Image} from 'react-native';
import { Card, CardItem, Thumbnail, Left, Body, Icon, Text, Header } from 'native-base';
// import { useFonts, Comfortaa_300Light } from '@expo-google-fonts/comfortaa';

import styles from '../style/style';



function PlantItem({ plantObj,navigation }) {
  // let [fontsLoaded] = useFonts({
  //   Comfortaa_300Light,
  // });
  return (
    <TouchableOpacity
      onPress={() => {
        navigation.navigate('PlantPage', {
          plantObj: plantObj
        })
      }}
    > 
         <Card style={{ elevation: 3 ,marginTop:'15%'}}>  
              <CardItem >
                 <Left>
                   <Thumbnail source={{uri:plantObj.imgUrl}} />
                   <Body>
                   <Text style={{fontFamily:'Comfortaa_600SemiBold'}}>{plantObj.name}</Text>
                 </Body>
               </Left>
               </CardItem>
              <CardItem cardBody>
                 <Image style={{ height: 350, flex: 1 }} source={{uri:plantObj.imgUrl}} />
               </CardItem>
               <CardItem>
            <Icon name="heart" style={{ color: '#ED4A6A' }} />
                <Body>
                  <Text style={{fontFamily:'Comfortaa_600SemiBold'}}>Family</Text>
                  <Text note style={{fontFamily:'Comfortaa_600SemiBold'}}>{plantObj.family}</Text>
                </Body>
                <Body>
                  <Text style={{fontFamily:'Comfortaa_600SemiBold'}}>Water</Text>
                  <Text note style={{fontFamily:'Comfortaa_600SemiBold'}}>{plantObj.waterAmount} per week</Text>
                </Body>
                <Body>
                  <Text style={{fontFamily:'Comfortaa_600SemiBold'}}>Temp</Text>
                  <Text note style={{fontFamily:'Comfortaa_600SemiBold'}}>{plantObj.recommendedTemp} </Text>
                </Body>
            </CardItem>
             </Card>
    </TouchableOpacity>
  );
}

PlantItem.propTypes = {
  plantObj: object
};

export default PlantItem;
