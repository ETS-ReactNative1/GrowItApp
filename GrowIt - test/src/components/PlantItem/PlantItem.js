import * as React from 'react';
import { object } from 'prop-types';
import { TouchableOpacity, Image} from 'react-native';
import { Card, CardItem, Thumbnail, Left, Body, Icon, Text, Header } from 'native-base';
import styles from './style';

function PlantItem({ plantObj,navigation }) {

  return (
    <TouchableOpacity
      onPress={() => {
        navigation.navigate('PlantPage', {
          plantObj: {...plantObj}
        })
      }}
    > 
         <Card style={styles.card}>  
              <CardItem >
                 <Left>
                   <Thumbnail source={{uri:plantObj.imgUrl}} />
                   <Body>
                   <Text style={styles.txt}>{plantObj.name}</Text>
                 </Body>
               </Left>
               </CardItem>
              <CardItem cardBody>
                 <Image style={styles.img} source={{uri:plantObj.imgUrl}} />
               </CardItem>
               <CardItem>
            <Icon name="heart" style={styles.icon} />
                <Body>
                  <Text style={styles.txt}>Family</Text>
                  <Text note style={styles.txt}>{plantObj.family}</Text>
                </Body>
                <Body>
                  <Text style={styles.txt}>Water</Text>
                  <Text note style={styles.txt}>each {plantObj.waterAmount/86400} days</Text>
                </Body>
                <Body>
                  <Text style={styles.txt}>Temp</Text>
                  <Text note style={styles.txt}>{plantObj.recommendedTemp} </Text>
                </Body>
            </CardItem>
             </Card>
    </TouchableOpacity>
  );
}

PlantItem.propTypes = {
  plantObj: object
};

export default PlantItem;
