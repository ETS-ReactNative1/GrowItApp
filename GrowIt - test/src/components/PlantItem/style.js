import { StyleSheet } from 'react-native'

const styles = StyleSheet.create({
    card: {
        elevation: 3,
        marginTop: '15%'
    },
    txt: {
        fontFamily: 'Comfortaa_600SemiBold'
    },
    img: {
        height: 350,
        flex: 1
    },
    icon: {
        color: '#ED4A6A'
    }

})

export default styles