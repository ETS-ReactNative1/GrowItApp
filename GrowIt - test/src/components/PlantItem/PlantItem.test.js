import React from 'react'
import { render } from '@testing-library/react-native'
import PlantItem from './PlantItem'

describe('PlantItem component Test', function () {
   
    test('add plant to PlantsList ', function () {
      const { baseElement }= render(<PlantItem key={1212} navigation={{}} plantObj={onePlant} />)
      expect(baseElement.toMatchSnapshot())
    });
  
});


const onePlant = {
    months: [7, 1, 8, 9, 2, 3, 4],
    _id: '5f0da4922f105dcd2e8b681c',
    name: 'Cucumber ',
    family: 'Cucurbitaceae',
    description: `Its elongated fruit is juicy and very rich in water. Its green color is eaten raw, pickled in salt or pickled.The plant originated in India, where humans began growing it about 3, 000 years ago.Today, cucumber is a very common vegetable, and it is grown in greenhouses or open fields.`,
    type: 'vegetables',
    waterAmount: '259200',
    imgUrl: 'https://img.wcdn.co.il/f_auto,w_1200,t_54/1/4/9/8/1498895-46.jpg',
    howToITreat:
      'Water the cucumbers until the soil is thoroughly absorbed and maintain this moisture at all times.',
    recommendedTemp: 25,
    recommendedHumidity: 25,
    recommendedClouds: 15
  }


  
