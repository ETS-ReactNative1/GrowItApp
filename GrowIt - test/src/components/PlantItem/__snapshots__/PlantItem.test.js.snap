// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlantItem component Test add plant to PlantsList  1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderBottomWidth": 0.5,
              "borderColor": "#ccc",
              "borderLeftWidth": 0.5,
              "borderRadius": 2,
              "borderRightWidth": 0.5,
              "borderTopWidth": 0.5,
              "elevation": 3,
              "flexWrap": "nowrap",
              "marginBottom": 5,
              "marginLeft": 2,
              "marginRight": 2,
              "marginTop": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 1.5,
            },
            Object {
              "elevation": 3,
              "marginTop": "15%",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "https://img.wcdn.co.il/f_auto,w_1200,t_54/1/4/9/8/1498895-46.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 28,
                  "height": 56,
                  "width": 56,
                }
              }
            />
            <View
              style={
                Object {
                  "alignItems": null,
                  "alignSelf": "center",
                  "flex": 1,
                  "marginLeft": 10,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    },
                    Object {
                      "fontFamily": "Comfortaa_600SemiBold",
                    },
                  ]
                }
                uppercase={false}
              >
                Cucumber 
              </Text>
            </View>
          </View>
        </View>
        <View
          cardBody={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": -5,
              "paddingLeft": -5,
              "paddingRight": -5,
              "paddingTop": -5,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://img.wcdn.co.il/f_auto,w_1200,t_54/1/4/9/8/1498895-46.jpg",
              }
            }
            style={
              Object {
                "flex": 1,
                "height": 350,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            }
          }
        >
          <Text />
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "alignSelf": "stretch",
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "fontFamily": "Comfortaa_600SemiBold",
                  },
                ]
              }
              uppercase={false}
            >
              Family
            </Text>
            <Text
              note={true}
              style={
                Array [
                  Object {
                    "color": "#808080",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "200",
                    "marginRight": 20,
                  },
                  Object {
                    "fontFamily": "Comfortaa_600SemiBold",
                  },
                ]
              }
              uppercase={false}
            >
              Cucurbitaceae
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "alignSelf": "stretch",
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "fontFamily": "Comfortaa_600SemiBold",
                  },
                ]
              }
              uppercase={false}
            >
              Water
            </Text>
            <Text
              note={true}
              style={
                Array [
                  Object {
                    "color": "#808080",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "200",
                    "marginRight": 20,
                  },
                  Object {
                    "fontFamily": "Comfortaa_600SemiBold",
                  },
                ]
              }
              uppercase={false}
            >
              each 
              3
               days
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "alignSelf": "stretch",
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "fontFamily": "Comfortaa_600SemiBold",
                  },
                ]
              }
              uppercase={false}
            >
              Temp
            </Text>
            <Text
              note={true}
              style={
                Array [
                  Object {
                    "color": "#808080",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "200",
                    "marginRight": 20,
                  },
                  Object {
                    "fontFamily": "Comfortaa_600SemiBold",
                  },
                ]
              }
              uppercase={false}
            >
              25
               
            </Text>
          </View>
        </View>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;

exports[`PlantItem component Test render plant item component snapshot  1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderBottomWidth": 0.5,
              "borderColor": "#ccc",
              "borderLeftWidth": 0.5,
              "borderRadius": 2,
              "borderRightWidth": 0.5,
              "borderTopWidth": 0.5,
              "elevation": 3,
              "flexWrap": "nowrap",
              "marginBottom": 5,
              "marginLeft": 2,
              "marginRight": 2,
              "marginTop": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 1.5,
            },
            Object {
              "elevation": 3,
              "marginTop": "15%",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "https://img.wcdn.co.il/f_auto,w_1200,t_54/1/4/9/8/1498895-46.jpg",
                }
              }
              style={
                Object {
                  "borderRadius": 28,
                  "height": 56,
                  "width": 56,
                }
              }
            />
            <View
              style={
                Object {
                  "alignItems": null,
                  "alignSelf": "center",
                  "flex": 1,
                  "marginLeft": 10,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 16,
                    },
                    Object {
                      "fontFamily": "Comfortaa_600SemiBold",
                    },
                  ]
                }
                uppercase={false}
              >
                Cucumber 
              </Text>
            </View>
          </View>
        </View>
        <View
          cardBody={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": -5,
              "paddingLeft": -5,
              "paddingRight": -5,
              "paddingTop": -5,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://img.wcdn.co.il/f_auto,w_1200,t_54/1/4/9/8/1498895-46.jpg",
              }
            }
            style={
              Object {
                "flex": 1,
                "height": 350,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            }
          }
        >
          <Text />
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "alignSelf": "stretch",
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "fontFamily": "Comfortaa_600SemiBold",
                  },
                ]
              }
              uppercase={false}
            >
              Family
            </Text>
            <Text
              note={true}
              style={
                Array [
                  Object {
                    "color": "#808080",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "200",
                    "marginRight": 20,
                  },
                  Object {
                    "fontFamily": "Comfortaa_600SemiBold",
                  },
                ]
              }
              uppercase={false}
            >
              Cucurbitaceae
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "alignSelf": "stretch",
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "fontFamily": "Comfortaa_600SemiBold",
                  },
                ]
              }
              uppercase={false}
            >
              Water
            </Text>
            <Text
              note={true}
              style={
                Array [
                  Object {
                    "color": "#808080",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "200",
                    "marginRight": 20,
                  },
                  Object {
                    "fontFamily": "Comfortaa_600SemiBold",
                  },
                ]
              }
              uppercase={false}
            >
              each 
              3
               days
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "alignSelf": "stretch",
                "flex": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "fontFamily": "Comfortaa_600SemiBold",
                  },
                ]
              }
              uppercase={false}
            >
              Temp
            </Text>
            <Text
              note={true}
              style={
                Array [
                  Object {
                    "color": "#808080",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "200",
                    "marginRight": 20,
                  },
                  Object {
                    "fontFamily": "Comfortaa_600SemiBold",
                  },
                ]
              }
              uppercase={false}
            >
              25
               
            </Text>
          </View>
        </View>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;
